packages/node_modules/@ciscospark/spark-core/src/config.js

```diff
@@ -4,71 +4,177 @@
+      */
+    redirect_uri: ['string', false, process.env.CISCOSPARK_REDIRECT_URI || process.env.COMMON_IDENTITY_REDIRECT_URI || process.env.REDIRECT_URI],
+
     /**
-     * {@see https://tools.ietf.org/html/rfc6749#section-4.1.4}
-     * @type {string}
-     */
-    scope: process.env.CISCOSPARK_SCOPE || process.env.CISCOSPARK_SCOPES || process.env.COMMON_IDENTITY_SCOPE || process.env.SCOPE,
+      * {@see https://tools.ietf.org/html/rfc6749#section-4.1.4}
+      * @type {string}
+      */
+    scope: ['string', false, process.env.CISCOSPARK_SCOPE || process.env.CISCOSPARK_SCOPES || process.env.COMMON_IDENTITY_SCOPE || process.env.SCOPE],
+
     /**
-     * Controls the UI of the CI login page.
-     * @private
-     * @type {string}
-     */
-    cisService: 'spark'
+      * Controls the UI of the CI login page.
+      * @private
+      * @type {string}
+      */
+    cisService: ['string', false, 'spark']
   },
+
+  derived: {
+    /**
+      * User activation URL
+      * {@link config.credentials.activationUrl}
+      * @type {string}
+      */
+    activationUrl: {
+      deps: ['idbroker.url'],
+      fn() {
+        return `${this.idbroker.url || 'https://idbroker.webex.com'}/idb/token/v1/actions/UserActivation/invoke`;
+      }
+    },
+
+
+    /**
+      * Authorization URL which prompts for user's password. Inferred from
+      * {@link config.credentials.authorizationString}
+      * @type {string}
+      */
+    authorizeUrl: {
+      deps: ['idbroker.url'],
+      fn() {
+        return process.env.AUTHORIZE_URL || `${this.idbroker.url}/idb/oauth2/v1/authorize`;
+      }
+    },
+
+    // TODO does hydra also have an access_token endpoint?
+    /**
+      * Token URL used for token refresh and auth code exchange
+      * @type {string}
+      */
+    tokenUrl: {
+      deps: ['idbroker.url'],
+      fn() {
+        return process.env.TOKEN_URL || `${this.idbroker.url}/idb/oauth2/v1/access_token`;
+      }
+    },
+
+    /**
+      * URL to revoke token
+      * @type {string}
+      */
+    revokeUrl: {
+      deps: ['idbroker.url'],
+      fn() {
+        return process.env.REVOKE_URL || `${this.idbroker.url}/idb/oauth2/v1/revoke`;
+      }
+    },
+
+    /**
+      * URL to load when the app logs out
+      * @type {string}
+      */
+    logoutUrl: {
+      deps: ['idbroker.url'],
+      fn() {
+        return `${this.idbroker.url}/idb/oauth2/v1/logout`;
+      }
+    },
+
+    /**
+      * Set password URL
+      * @type {string}
+      */
+    setPasswordUrl: {
+      deps: ['identity.url'],
+      fn() {
+        return `${this.identity.url || 'https://identity.webex.com'}/identity/scim/v1/Users`;
+      }
+    }
+
+  }
+
+});
+
+
+export default {
+  maxAppLevelRedirects: 10,
+  maxLocusRedirects: 5,
+  maxAuthenticationReplays: 1,
+  maxReconnectAttempts: 1,
+  onBeforeLogout: [],
+  trackingIdPrefix: 'spark-js-sdk',
+  trackingIdSuffix: '',
+  AlternateLogger: undefined,
+  credentials: new CredentialInfo({
```

Comment:
Good question.  Looking at the examples in the docs, the children are Ampersand objects:
https://ampersandjs.com/docs/#ampersand-state-children
In the SDK, SparkInternalCore in property internal is also an Ampersand object.
So it would seem having the child be an Ampersand object is the model.
Regardless it raises the question whether config should be changed to a child property too (like SparkInternalCore).  It's currently still a session property.  I will look into that.

Reply:
I don't think we want to declare config as a child.  This will mean (config change) events are propagated up which may create unnecessary noise.